hwmv2:  qemu_x86: acrn: Port to HWMv2

golowanow opened the pull request:
Port qemu_x86 to HWMv2 as well as SoCs it depends on:
* ia32
* lakemont


Several qemu_x86 board configurations have changed their identifiers to the HWMv2 scheme:

 - `qemu_x86_nokpti` --> `qemu_x86/ia32/nokpti`
 - `qemu_x86_nommu`  --> `qemu_x86/ia32/nommu`
 - `qemu_x86_nopae`  --> `qemu_x86/ia32/nopae`
 - `qemu_x86_virt`   --> `qemu_x86/ia32/virt`
 - `qemu_x86_xip`    --> `qemu_x86/ia32/xip`
 - `qemu_x86_64_nokpti` --> `qemu_x86/ia32/nokpti`
 - `qemu_x86_tiny@768` --> `qemu_x86_tiny/ia32/768`

Also `x86/acrn` boards: `acrn` and `acrn_ehl_crb`

golowanow commented:
hmm, is `qemu_x86_virt` (and Co.) a board itself, or just a revision of `qemu_x86@virt` ?
currently west accepts both, but twister only as a board.

golowanow committed:
tests: kernel: qemu_x86: adjust to the HWMv2

Adjust qemu_x86 board configuration names to HWMv2 scheme.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow commented:
@nashif, is it known and considered acceptable: with the HWMv2 board names Twister creates additional levels of directory structure under `./twister-out` for  `board/soc/revision`  instead of just one old `board_soc_revision`, for example after `./scripts/twister -vv -a x86 -T tests/arch/x86/pagetables/ ` it looks like below
```
./twister-out
   |-qemu_x86
   |---ia32
   |-----nokpti
   |-------tests
   |---------arch
   |-----------x86
   |-------------pagetables
   |---------------arch.x86.pagetables
...
```

nashif commented:
> @nashif, is it known and considered acceptable: with the HWMv2 board names Twister creates additional levels of directory structure under `./twister-out` for `board/soc/revision` instead of just one old `board_soc_revision`, for example after `./scripts/twister -vv -a x86 -T tests/arch/x86/pagetables/ ` it looks like below
> 
> ```
> ./twister-out
>    |-qemu_x86
>    |---ia32
>    |-----nokpti
>    |-------tests
>    |---------arch
>    |-----------x86
>    |-------------pagetables
>    |---------------arch.x86.pagetables
> ...
> ```

No, was not aware of this.

This is a mess :( I do not like the slash in the board names and IMO this is going to be a problem and a blocker. FYI @tejlmand

golowanow commented:
two problems with CI twister tests currently on this PR:
1) `qemu_xtensa` `tests/subsys/llext/hello_world/llext.simple.modules_enabled` - not related to this change.
2) old `x86\acrn` can't run properly as it depends on `soc/v2/ia32` here and `soc/x86/elkhart_lake` which is in transition at #67885.
 I think acrn might be moved as a separate PR when both SoC it uses migrate to the `collab-hwm`, or at least one of them.

golowanow commented:
@tejlmand - could you please provide some feedback - is this PR on track with the general v1 to v2 migration principles ?

nashif commented:
regarding twister, would be nice to patch twister to replace `/` with `_` in the build_dir to restore existing functionality and tree structure.

tejlmand commented on boards/v2/qemu/qemu_x86/board.yml:
```diff
@@ -0,0 +1,30 @@
+    socs:
+      - name: ia32
+    revision:
+      format: custom
```
with hwmv2 it is now possible to actually implement a board revision check here: https://github.com/zephyrproject-rtos/zephyr/blob/dcd31306f113c1183d0c7d497921d1dd195a9c93/boards/v2/qemu/qemu_x86/revision.cmake#L4-L5 because only boards which specifies `revision: format: custom` will now load this file.

All other boards in this folder / yaml file will ignore the `revision.cmake` file.

Note, this comment is not blocking for this PR, more an extra piece of info.

tejlmand commented on boards/v2/qemu/qemu_x86/board.yml:
```diff
@@ -0,0 +1,30 @@
+      format: custom
+      default: ''
+      revisions:
+        - name: '768'
```
if revisions are numbers, then why not  use `revision: format: number` ?

nordicjm replied:
This should be a variant then, not a revision

golowanow replied:
changed to a variant

tejlmand reviewed (commented):
looks very nice.
Only minor nits observed.

tejlmand commented:
> This is a mess :( I do not like the slash in the board names and IMO this is going to be a problem and a blocker. FYI @tejlmand

should be as simple as having twister sanitize the name before creating the folder structure, but would need to see if twister in other places than folder creation is also constructing the path based on board names.

tejlmand commented:
> should be as simple as having twister sanitize the name before creating the folder structure, but would need to see if twister in other places than folder creation is also constructing the path based on board names.

added as open task here: https://github.com/zephyrproject-rtos/zephyr/issues/51831

tejlmand reviewed (approved):
approved, nits can be addressed in followup.

nordicjm commented on boards/v2/qemu/qemu_x86/Kconfig.defconfig:
```diff
@@ -46,6 +55,9 @@ if BOARD_QEMU_X86_64
 config BOARD
 	default "qemu_x86_64"
```
this should be removed

golowanow replied:
done

nordicjm commented on boards/v2/qemu/qemu_x86/Kconfig.defconfig:
```diff
@@ -10,13 +10,22 @@ config EEPROM_INIT_PRIORITY
 
 if BOARD_QEMU_X86
 
 config BOARD
 	default "qemu_x86"
```
And this, etc. throughout file

golowanow replied:
done

nordicjm commented on boards/v2/qemu/qemu_x86/qemu_x86_64_nokpti.yaml:
```diff
@@ -1,4 +1,4 @@
-identifier: qemu_x86_64_nokpti
+identifier: qemu_x86_64//nokpti
```
this should use the full identifier

nordicjm commented on boards/v2/qemu/qemu_x86/qemu_x86_nokpti.yaml:
```diff
@@ -1,4 +1,4 @@
-identifier: qemu_x86_nokpti
+identifier: qemu_x86/ia32/nokpti
```
as above

golowanow replied:
> as above

@nordicjm,  isn't `qemu_x86/ia32/nokpti` a full identifier already ?

nordicjm replied:
yes, marked as resolved for this

nordicjm commented on boards/v2/qemu/qemu_x86/Kconfig.defconfig:
```diff
@@ -46,6 +55,9 @@ if BOARD_QEMU_X86_64
 config BOARD
 	default "qemu_x86_64"
 
+config X86_64
+	default y
```
(non-blocking) would be far more concise to have these in a `Kconfig` file

golowanow replied:
done

golowanow commented on soc/v2/intel/lakemont/Kconfig.soc:
```diff
@@ -0,0 +1,5 @@
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_LAKEMONT
+	bool
```
done

nordicjm commented on boards/v2/qemu/qemu_x86/Kconfig.qemu_x86_lakemont:
```diff
@@ -0,0 +1,7 @@
+
+config BOARD_QEMU_X86_LAKEMONT
+	select SOC_LAKEMONT
+	select CPU_HAS_FPU
```
As above

golowanow replied:
done

golowanow commented on boards/v2/qemu/qemu_x86/Kconfig.qemu_x86:
```diff
@@ -0,0 +1,7 @@
+
+config BOARD_QEMU_X86
+	select SOC_IA32
+	select CPU_HAS_FPU
```
> CPU_HAS_FPU should be in `Kconfig` not `Kconfig.<x>`

(confused) why ?

nordicjm replied:
`Kconfig.<board>` is used by sysbuild and twister, it should be selecting things for filtering namely the SOC, analogous to the `Kconfig.soc` file in the soc folder whereas the `Kconfig` file is analogous to the `Kconfig` file in the soc folder. This is why `Kconfig.soc` does things like sets the family, series and soc itself but does not specify the features of the soc

golowanow replied:
thanks, done

golowanow commented on soc/v2/intel/lakemont/Kconfig.soc:
```diff
@@ -0,0 +1,8 @@
+	bool
+
+config SOC
+  default "lakemont" if SOC_LAKEMONT
```
done

golowanow commented on soc/v2/ia32/Kconfig.soc:
```diff
@@ -0,0 +1,9 @@
+	bool
+
+config SOC
+  default "ia32" if SOC_IA32
```
done

golowanow commented on boards/v2/qemu/qemu_x86/Kconfig.qemu_x86_64:
```diff
@@ -0,0 +1,7 @@
+
+config BOARD_QEMU_X86_64
+	select SOC_IA32
+	select X86_64
```
done

golowanow commented on boards/v2/qemu/qemu_x86/Kconfig.qemu_x86_tiny:
```diff
@@ -0,0 +1,7 @@
+
+config BOARD_QEMU_X86_TINY
+	select SOC_IA32
+	select CPU_HAS_FPU
```
done

golowanow committed:
soc: x86: lakemont: move and convert to HWMv2

Move and convert soc/x86/lakemont to HWMv2

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
soc: x86: ia32: move and convert to HWMv2

Move and convert soc/x86/ia32 to HWMv2.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
boards: x86: qemu_x86: move and convert to HWMv2

Move and convert qemu_x86 board configurations to HWMv2.

Several board configurations have changed their identifiers
to the HWMv2 scheme:

 - qemu_x86_nokpti --> qemu_x86/ia32/nokpti
 - qemu_x86_nommu  --> qemu_x86/ia32/nommu
 - qemu_x86_nopae  --> qemu_x86/ia32/nopae
 - qemu_x86_virt   --> qemu_x86/ia32/virt
 - qemu_x86_xip    --> qemu_x86/ia32/xip

 - qemu_x86_64_nokpti --> qemu_x86_64/ia32/nokpti

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
boards: x86: qemu_x86_tiny@768: change to board variant

Change `qemu_x86_tiny@768` board revision to `qemu_x86_tiny/ia32/768`
board variant configuration.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow commented on boards/v2/qemu/qemu_x86/qemu_x86_nommu_defconfig:
```diff
@@ -1,7 +1,5 @@
 # SPDX-License-Identifier: Apache-2.0
 
-CONFIG_SOC_IA32=y
```
thank you, file names are adjusted

golowanow commented:
1. adjusted to #68346 
2. review comments addressed

golowanow committed:
boards: x86: qemu_x86: optimize default HWMv2 configurations

Remove duplicated Kconfig default options from qemu_x86 board
variants' configuration keeping only differences in regard
of the appropriate board configurations.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

nordicjm commented on boards/qemu/qemu_x86/board.yml:
```diff
@@ -23,8 +23,5 @@ boards:
-    revision:
-      format: custom
-      default: ''
-      revisions:
```
These changes can be merged into the main commit

golowanow replied:
> These changes can be merged into the main commit

? this piece of code is already outdated

golowanow replied:
imo its better to keep this commit separate to track the board naming change
(`qemu_x86_tiny@768` board revision to `qemu_x86_tiny/ia32/768`)

nordicjm replied:
For future going forward it should just be part of the original commit, these commits will only exist in the collab-hwm branch, when it is ready for merging into `main` every commit is going to be squashed into one and doing it like above just causes confusion for the reviewer

golowanow replied:
> For future going forward it should just be part of the original commit, these commits will only exist in the collab-hwm branch, when it is ready for merging into `main` every commit is going to be squashed into one and doing it like above just causes confusion for the reviewer

quemux86_tiny@768 rename caused by HWMv2 migration is just one of 6 commits here which all be squashed later anyway. What other commit it have to be a part of now ?

golowanow commented:
@nordicjm - are any review blockers still remaining here ?

golowanow commented:
> @nordicjm - are any review blockers still remaining here ?

@nordicjm, @nashif, @57300, @tejlmand @jhedberg @tbursztyka - up to your attention

nordicjm commented on tests/kernel/threads/dynamic_thread_stack/testcase.yaml:
```diff
@@ -3,7 +3,7 @@ common:
   min_ram: 32
   integration_platforms:
     - qemu_x86
```
@tejlmand should these be updated also since they have variants and the underlying board.yml has the full name, or is it okay to just keep the shortened name here?

golowanow replied:
twister currently takes `qemu_x86` as `qemu_x86/ia32` and other variants are named explicitly, aren't they ?
```
CONFIG_BOARD="qemu_x86"                                                                                                                                                                                                                                               CONFIG_BOARD_REVISION=""                                                                                                                                                                                                                                              CONFIG_BOARD_QEMU_X86=y                                                                                                                                                                                                                                               CONFIG_BOARD_QEMU_X86_IA32=y      
```

nordicjm commented:
Needs rebase, and up squared 7000 board is failing CI: https://github.com/zephyrproject-rtos/zephyr/actions/runs/7843368285/job/21403615547?pr=68801

golowanow commented:
> Needs rebase, 

rebased, now it needs #68801

golowanow commented:
> Needs rebase, and up squared 7000 board is failing CI: https://github.com/zephyrproject-rtos/zephyr/actions/runs/7843368285/job/21403615547?pr=68801

@nordicjm it seems `up_squared_pro_7000` is failing because changes from #62694 were not applied to the boards migrated with `collab_hwm` branch on its rebase. 
I'll prepare a fix for the intel boards, but isn't something wrong with that 'v2' / 'legacy' bulk move ?
@tejlmand

golowanow commented:
@nordicjm do you have objections to merge this PR ?
* compliance check passing if #68811 applied
* twister run issues are not related to this PR

nordicjm commented:
> @nordicjm do you have objections to merge this PR ?
> 
>     * compliance check passing if [[hwm2] ci: update check_compliance to not create duplicate lines in Kconfig #68811](https://github.com/zephyrproject-rtos/zephyr/pull/68811) applied
> 
>     * twister run issues are not related to this PR

Needs another approval

golowanow commented:
> Needs another approval

@jhedberg, @tejlmand, @nashif could you please look at this PR once more ?

golowanow commented:
rebased to last changes at `collab-hwm`

nordicjm reviewed (changes requested):
`acrn` board needs fixing

golowanow commented:
> `acrn` board needs fixing

@nordicjm , it is planned as separate PR to follow on this one.

https://github.com/zephyrproject-rtos/zephyr/pull/68078#issuecomment-1910834049

> 2. old `x86\acrn` can't run properly as it depends on `soc/v2/ia32` here and `soc/x86/elkhart_lake` which is in transition at [hwmv2: port intel x86 boards #67885](https://github.com/zephyrproject-rtos/zephyr/pull/67885).

nordicjm commented:
> > `acrn` board needs fixing
> 
> @nordicjm , it is planned as separate PR to follow on this one.
> 
> [#68078 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/68078#issuecomment-1910834049)
> 
> > 2. old `x86\acrn` can't run properly as it depends on `soc/v2/ia32` here and `soc/x86/elkhart_lake` which is in transition at [hwmv2: port intel x86 boards #67885](https://github.com/zephyrproject-rtos/zephyr/pull/67885).

Can you put the commits here? Trying to fix up CI on collab-hwm not make it worse

golowanow committed:
boards: x86: acrn: move and convert to HWMv2

Move and convert to HWMv2 `acrn` and `acrn_ehl_crb` board
configurations.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow commented:
> > > `acrn` board needs fixing
> > 
> > 
> > @nordicjm , it is planned as separate PR to follow on this one.
> > [#68078 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/68078#issuecomment-1910834049)
> > > 2. old `x86\acrn` can't run properly as it depends on `soc/v2/ia32` here and `soc/x86/elkhart_lake` which is in transition at [hwmv2: port intel x86 boards #67885](https://github.com/zephyrproject-rtos/zephyr/pull/67885).
> 
> Can you put the commits here? Trying to fix up CI on collab-hwm not make it worse

@nordicjm, done, `x86/acrn` is also here.
